cluster/cluster.go

```diff
@@ -217,6 +217,9 @@ func (p *Peer) Join(
 	n, err := p.mlist.Join(p.resolvedPeers)
 	if err != nil {
 		level.Warn(p.logger).Log("msg", "failed to join cluster", "err", err)
+		if reconnectInterval != 0 {
```

Comment:
Nit: might be clearer to say “will retry joining cluster in %v” otherwise it sounds a little funny IMO

Reply:
"will retry joining cluster in" makes it sound like a one-shot, while this is repeated :man_shrugging:

Reply:
I am fine with either way, but I would like to include the fact that this is a periodic action. @squat how can we combine the two?

Reply:
“Retrying every” to me implies that this is a permanent state, not just on failure. In that case, the correct place for this is right after the connection failure. That way the line is logged after every failure, e.g.:
```console
failed to join cluster
retrying in %v
failed to join cluster
retrying in %v
```

Reply:
Again it’s a nit, but sounds odd to me as a native English speaker. Not a blocker by any means.

Reply:
> that this is a permanent state, not just on failure.

Discussed offline with @squat, as this  is indeed a permanent periodic process we can move on.